src/Arena.js

```diff
@@ -1,554 +1,555 @@
-import React, { useEffect } from "react";
+import React, { useEffect, useState } from "react";
 import ReactTooltip from "react-tooltip";
-import {
-  Route,
-  NavLink,
-} from "react-router-dom";
+import { Route, NavLink } from "react-router-dom";
 import Staging from "./Staging";
-import { useSelector, useDispatch } from 'react-redux';
-import { attack, critAttack, riposte, opponentAttack, 
-          blockedAttack, resetHealth, endTurn, skillBuff, 
-          skill, setSkill, opponentDisabled, poisonDmg, bleedDmg} from './actions'
-import { diceRoll, sleep, fadeOut, applyPoisonStyle, oppGetHurt, dodgeShake, oppAttackAnimation, closeDropup, openDropup, bleedTickAnimation, animatePlayerAttack, animateDualWieldAttack, displayDmgNumber, animateBuff, addActiveBuff} from './data/functions'
-import ClickNHold from 'react-click-n-hold';
+import { useSelector, useDispatch } from "react-redux";
+import {
+  attack,
+  critAttack,
+  riposte,
+  opponentAttack,
+  blockedAttack,
+  resetHealth,
+  endTurn,
+  skillBuff,
+  skill,
+  setSkill,
+  opponentDisabled,
+  poisonDmg,
+  bleedDmg,
+} from "./actions";
+import {
+  diceRoll,
+  sleep,
+  applyPoisonStyle,
+  oppGetHurt,
+  dodgeShake,
+  oppAttackAnimation,
+  closeDropup,
+  openDropup,
+  bleedTickAnimation,
+  displayDmgNumber,
+  animateBuff
+} from "./data/functions";
+import ClickNHold from "react-click-n-hold";
+import Gladiator from "./models/Gladiator";
+import Opponent from "./models/Opponent";
+
+// PNG assets
+import riposteImage from './assets/images/animations/riposte.gif';
+import { OpponentType } from "./data/opponentData";
 
-let stack = 0
 
 export default function Arena() {
-  
   const dispatch = useDispatch();
-  const battleState = useSelector(state => state.BattleState)
-  const myGladiator = useSelector(state => state.ChosenGladiator.chosen)
-  const opponent = useSelector(state => state.ChosenOpponent.chosenOpp)
-  const opponentArray = useSelector(state => state.OpponentReducer)
-  const allSkills = useSelector(state => state.SkillsReducer)
-  let mySkill = useSelector(state => state.SelectedSkill.mySkill)
-  let skillArray = [];
-  let activeBuffs = battleState.activeBuffs;
+  const {
```

Comment:
Here's a nice technique called destructuring where you can declare variables that are inside the right hand expression at different levels, so you only have to call useSelector once